### PR TITLE
feat: add booking preview page

### PR DESCRIPTION
**Pull request by eugenederii**

### CODE REVIEW

**Comment by VadimTarasenko** [comment]
Screenshot 2024-07-03 at 23 19 58

@eugenederii Getting this error when trying to book service. Please find the reason of issue. Reply here please.

**Comment by VadimTarasenko** [comment]
@eugenederii Research how to implement "Add to calendar". It should generate event file `.ics` and download right after file is ready. Display spinner in button until file is ready.

**Comment by VadimTarasenko** [comment]
Screenshot 2024-07-03 at 23 32 21

Remove this title for now.

**Comment by VadimTarasenko** [comment]
Screenshot 2024-07-03 at 23 32 51

Contact button isn't working. And popup width should fit the content.

**Comment by VadimTarasenko** [comment]
When click on avatar or username - open user profile page in the new tab.

Screenshot 2024-07-10 at 19 05 15

**Comment by VadimTarasenko** [comment]
.ics file isn't valid. Getting this result when opening downloaded file.

Screenshot 2024-07-10 at 19 09 12

**Comment by VadimTarasenko** [comment]
@eugenederii please fix conflicts

File: src/app/[lang]/bookings/[code]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+'use client';
+
+// containers
+import { BookingPreview } from '@/modules/booking/containers/booking-preview/booking-preview';
```
Comment (VadimTarasenko) [inline src/app/[lang]/bookings/[code]/page.tsx:4]: create index.ts file

File: src/modules/booking/components/booking-preview-detail-box/booking-preview-detail-box.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { type FC } from 'react';
+
+import { Typography } from '@/modules/core/components/typogrpahy';
+import { Avatar } from '@/modules/core/components/avatar';
+
+import type { BookingDetailProps } from './booking-preview-detail-box.interface';
+
+export const BookingDetailBox: FC<BookingDetailProps> = ({
```
Comment (VadimTarasenko) [inline src/modules/booking/components/booking-preview-detail-box/booking-preview-detail-box.tsx:8]: the name of component should be same as file name

File: src/modules/booking/components/modal-success/modal-successs.tsx [hunk 3]
```diff
@@ -97,7 +100,9 @@ export const BookingModalSuccess: FC<BookingModalSuccessProps> = ({
             className={styles.closeBtn}
           />
           <Button
+            onClick={() => router.push(`/bookings/${bookingData?.code}`)}
```
Comment (VadimTarasenko) [inline src/modules/booking/components/modal-success/modal-successs.tsx:103]: `onClick={() => window.open(`/bookings/${bookingData?.code}`, '_blank')}`

File: src/modules/booking/containers/booking-preview/booking-preview.tsx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  // queries
+  const bookingDetails = trpc.booking.getByCode.useQuery({
+    code,
+    expand: ['serviceProfessional'],
+  });
+
+  const professional = trpc.professional.get.useQuery(
+    {
+      // @ts-ignore
+      id: bookingDetails.data?.serviceProfessional?.professional?.userId || '',
```
Comment (VadimTarasenko) [inline src/modules/booking/containers/booking-preview/booking-preview.tsx:73]: @IhorBalan  `professional` object exist in the response but doesn't appear in types

File: src/modules/booking/containers/booking-preview/booking-preview.tsx [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+          if (link !== null) {
+            window.open(link, '__blank');
+          }
+        }}
+        className='mt-12'
+        variant='outlined'
+        text='Open on google maps'
+      />
+    </main>
+  );
```
Comment (VadimTarasenko) [inline src/modules/booking/containers/booking-preview/booking-preview.tsx:327]: @eugenederii Please divide this page on smaller components

File: src/app/[lang]/bookings/[code]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+'use client';
```
Comment (VadimTarasenko) [inline src/app/[lang]/bookings/[code]/page.tsx:1]: remove this 'use client'

**Review by VadimTarasenko — APPROVED** [review]

**Review by IhorBalan — APPROVED** [review]

**Pull request merged**